src/common/util.js

```diff
@@ -401,7 +401,7 @@ var Cookies = {
       return loginids.map(function(_id) { return _id.id }).indexOf(id.id) === -1;
     });
 
-    return oauth_loginids ? oauth_loginids : loginids
+    return oauth_loginids.length > 0 ? oauth_loginids : loginids
```

Comment:
@arnabk Yes, I did. Although I think I should add the extra check, in order to be certain that it never breaks, even if someone changes the statement above.